tiledb/sm/cpp_api/filter_experimental.h

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file filter_experimental.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the experimental C++ API for filters.
+ */
+#ifndef TILEDB_CPP_API_ARRAY_FILTER_H
+#define TILEDB_CPP_API_ARRAY_FILTER_H
+
+#include <vector>
+#include "filter.h"
+#include "tiledb.h"
+
+namespace tiledb {
+
+/**
+ * Returns the datatype of the filter option.
+ */
+tiledb_datatype_t filter_option_datatype(tiledb_filter_option_t option);
```

Comment:
I don't know if we have any other standalone helper functions like this. I couldn't find any, but this doesn't seem like it belongs on the Filter class.